src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,318 @@
+import React, {
+  forwardRef,
+  useEffect,
+  useState,
+  useRef,
+  CSSProperties,
+  useCallback,
+} from "react";
+import {
+  Box,
+  chakra,
+  ChakraComponent,
+  useMultiStyleConfig,
+  Flex,
+} from "@chakra-ui/react";
+import Button from "../Button/Button";
+import Link from "../Link/Link";
+import List from "../List/List";
+
+export const actionBackgroundColorsArray = [
+  "brand.primary-05",
+  "section.blogs.primary-05",
+  "section.books-and-more.primary-05",
+  "section.connect.primary-05",
+  "section.education.primary-05",
+  "section.locations.primary-05",
+  "section.research.primary-05",
+  "section.research-library.lpa-05",
+  "section.research-library.schomburg-05",
+  "section.research-library.schwarzman-05",
+  "section.whats-on.primary-05",
+  "dark.brand.primary-05",
+  "dark.section.blogs.primary-05",
+  "dark.section.books-and-more.primary-05",
+  "dark.section.connect.primary-05",
+  "dark.section.education.primary-05",
+  "dark.section.locations.primary-05",
+  "dark.section.research.secondary-05",
+  "dark.section.research-library.lpa-05",
+  "dark.section.research-library.schomburg-05",
+  "dark.section.research-library.schwarzman-05",
+  "dark.section.whats-on.primary-05",
+] as const;
+
+export type actionBackgroundColors = typeof actionBackgroundColorsArray[number];
+
+export const highlightColorsArray = [
+  "brand.primary",
+  "section.blogs.primary",
+  "section.books-and-more.primary",
+  "section.connect.primary",
+  "section.education.primary",
+  "section.locations.primary",
+  "section.research.primary",
+  "section.research-library-lpa.primary",
+  "section.research-library-schomburg.primary",
+  "section.research-library-schwarzman.primary",
+  "section.whats-on.primary",
+  "dark.brand.primary",
+  "dark.section.blogs.primary",
+  "dark.section.books-and-more.primary",
+  "dark.section.connect.primary",
+  "dark.section.education.primary",
+  "dark.section.locations.primary",
+  "dark.section.research.secondary",
+  "dark.section.research-library-lpa.primary",
+  "dark.section.research-library-schomburg.primary",
+  "dark.section.research-library-schwarzman.primary",
+  "dark.section.whats-on.primary",
+];
+export type highlightColors = typeof highlightColorsArray[number];
+
+export interface SubNavProps {
+  /** Additional class name for the `SubNav` component. */
+  className?: string;
+  /**
+   * Primary actions displayed on the left side of the SubNav.
+   * Use SubNavButton and SubNavLink components, which mirror
+   * the DS Button and Link.
+   */
+  primaryActions: React.ReactNode;
```

Comment:
Maybe we should go back to the idea of this being an array of buttons or links. Then we can pass the selectedItem into the SubNav (parent) and then pass an isSelected prop into the button/link with a matching id. @EdwinGuzman, was your issue with this idea that you didn't want to loop through them?